flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -654,11 +654,7 @@ private void callSelect(SqlCommandCall cmdCall) {
         if (resultDesc.isTableauMode()) {
             try (CliTableauResultView tableauResultView =
                     new CliTableauResultView(terminal, executor, sessionId, resultDesc)) {
-                if (resultDesc.isMaterialized()) {
-                    tableauResultView.displayBatchResults();
-                } else {
-                    tableauResultView.displayStreamResults();
-                }
+                tableauResultView.displayResults(resultDesc.isStreamingMode());
```

Comment:
The `CliTableauResultView` already holds the `resultDesc`, so there is no need to pass the streaming mode again.